site/content/docs/howto/configure-auth-for-webapps.md

```diff
@@ -0,0 +1,347 @@
+```
+
+Do not share OIDCClients between multiple web applications. Each web application should have its own OIDCClient.
+
+The `name` of the OIDCClient will be the client ID used by the web application in the OIDC flows.
+
+The `allowedGrantTypes` and `allowedScopes` decides what the web application is allowed to do with respect to
+authentication. There are several typical combinations of these settings:
+
+1. A web application which is allowed to use the Supervisor for authentication, and furthermore is allowed to
+   authenticate into Kubernetes clusters and perform actions on behalf of the users (using the user's identity):
+
+    ```yaml
+      allowedGrantTypes:
+        - authorization_code
+        - refresh_token
+        - urn:ietf:params:oauth:grant-type:token-exchange
+      allowedScopes:
+        - openid
+        - offline_access
+        - pinniped:request-audience
+        - username
+        - groups
+    ```
+
+2. A web application which is allowed to use the Supervisor for authentication, but cannot perform actions on
+   Kubernetes clusters.
+
+    ```yaml
+      allowedGrantTypes:
+        - authorization_code
+        - refresh_token
+      allowedScopes:
+        - openid
+        - offline_access
+        - username
+        # "groups" can be excluded from this list when the webapp does
+        # not need to see the group memberships of the users.
+        - groups 
+    ```
+
+3. A web application which is allowed to use the Supervisor for authentication, but cannot see the username or
+   group memberships of the authenticated users, and cannot perform actions on Kubernetes clusters.
+
+    ```yaml
+      allowedGrantTypes:
+        - authorization_code
+        - refresh_token
+      allowedScopes:
+        - openid
+        - offline_access
+    ```
+
+## Create a client secret for the OIDCClient
+
+For each OIDCClient created by the Supervisor administrator, the administrator will also need to generate a client
+secret for the client. The client secrets are random strings auto-generated by the Supervisor upon request.
+The plaintext secret will only be returned once upon creation.
+
+```sh
+cat <<EOF | kubectl create -f -
+apiVersion: clientsecret.supervisor.pinniped.dev/v1alpha1
+kind: OIDCClientSecretRequest
+metadata:
+  name: client.oauth.pinniped.dev-my-webapp-client # the name of the OIDCClient
+  namespace: supervisor # the namespace of the OIDCClient
+spec:
+  generateNewSecret: true
+EOF
+```
+
+The server will respond with the newly generated client secret, e.g.:
+
+```
+NAMESPACE               NAME                                        SECRET     TOTAL
+pinniped-supervisor     client.oauth.pinniped.dev-my-webapp-client  abc123     1
+```
+
+Take care to make a note of the secret. After it has been returned by the create API, there is no other way to
+retrieve it in the future. The secret is not stored in plaintext on the server, which only stores a
+bcrypt-hashed version of the secret.
+
```

Comment:
Also talk about how the other `kubectl` verbs will not work. Only create works for this API type. Can't get, get by name, delete, etc.